whipper/common/encode.py

```diff
@@ -89,3 +89,62 @@ def _tag(self):
         w.save()
 
         self.stop()
+
+
+class EmbedPictureTask(task.Task):
+    description = 'Embed picture to FLAC'
+
+    def __init__(self, track_path, coverart_path):
+        self.track_path = track_path
+        self.coverart_path = coverart_path
+
+    def start(self, runner):
+        task.Task.start(self, runner)
+        self.schedule(0.0, self._embed_picture)
+
+    def _make_flac_picture(self, coverart_filename):
+        """Given a path to a jpeg or png file, return a flac.Picture
+
+        Returns: a valid flac.Picture or None
+        """
+        if not coverart_filename:
+            return None
+
+        try:
+            from PIL import Image
+        except ImportError as e:
+            logger.warning("pillow is required to grab cover art: %r", e)
+
+        im = Image.open(coverart_filename)
+        if im.format == 'JPEG':
+            mime = 'image/jpeg'
+        elif im.format == 'PNG':
+            mime = 'image/png'
+        else:
+            return None          # we only support png and jpeg
```

Comment:
I think you should add a logger statement before the `return` to explain that no cover art will be added because the fetched image is in an unsupported format (even though the correct option has been provided). Another possibility, even if probably less reliable (can be wrong), would be to check, in the `getCoverArt` method, that the `content-type` header of the requested file is equal to `image/jpeg` or `image/png`.

Reply:
The logger statement is added. I'm not sure about the second suggestion tho.

Reply:
> I'm not sure about the second suggestion tho.

I've added an example in another comment. :wink: